### PR TITLE
Add DestructuringDeclarationWithTooManyEntries rule

### PR DESCRIPTION
**Pull request by 3flex**
This rule checks for too many destructuring entries when using destructuring declarations. I've seen IntelliJ in the past recommend using destructuring on data classes with IMHO far too many fields, making the suggested code harder to understand. It's also [suggested by Kotlin's team](https://www.reddit.com/r/Kotlin/comments/ji9z19/kotlin_team_ama_2_ask_us_anything/ga5abq3/) that destructuring be used mostly for things like key/value pairs with only a very small number of elements.

The default maximum of 3 was chosen because it allows easy use of Kotlin stdlib `Pair` and `Triple`, and 4 seemed too many.

There weren't any violations in detekt's codebase when I was testing :)

### CODE REVIEW

**Comment by BraisGabin** [comment]
I love it! We clearly need rules about destructuring declarations. Could you extend this issue for destructuring in lambda paramenters? 

```kotlin
hello { (_, name) ->
  println("Hello $name")
}
```

These parenthesis are nearly invisible for some people because in java that those parenthesis do nothing.

File: detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/DestructuringDeclarationWithTooManyEntriesSpec.kt [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                    println(x)
+                    println(y)
+                    
+                    val (a, b, c) = Triple(1, 2, 3)
+                    println(a)
+                    println(b)
+                    println(c)
+                }
+            """.trimIndent()
+                assertThat(subject.lint(code)).isEmpty()
```
Comment (cortinico) [inline detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/DestructuringDeclarationWithTooManyEntriesSpec.kt:29]: Please replace with `compileAndLint` or `compileAndLintWithContext`

**Review by cortinico — APPROVED** [review]
Great addition 🚀

**Review by schalkms — COMMENTED** [review]
I like this new rule and would like to have it in the next release. What’s missing to get this merged?

**Comment by 3flex** [comment]
@BraisGabin 

> Could you extend this issue for destructuring in lambda paramenters?

The rule picks these up since the compiler treats destructuring declarations in lambdas the same way as when using `val (x, y) = ...`. I've added a couple of tests though to show this.

This should be ready to merge now.

**Review by BraisGabin — APPROVED** [review]

**Pull request merged**